packages/palette/src/elements/ShowMore/ShowMore.tsx

```diff
@@ -14,23 +16,27 @@ export const ShowMore: React.FC<ShowMoreProps> = ({
   initial = INITIAL_ITEMS_TO_SHOW,
   children,
   expanded = false,
+  hideText = "Hide",
+  showMoreText = "Show more",
+  ...rest
 }) => {
   const [isExpanded, setExpanded] = useState(expanded)
   const nodes = Children.toArray(children).filter(isValidElement)
   const hasMore = nodes.length > initial
 
+  const [textProps] = splitTextProps(rest)
+  const [clickableProps] = splitClickableProps(rest)
```

Comment:
@dzucconi how do you feel about this?

I did try an approach where `ShowMore` just had a `<div>` instead of `<Text>` so it could inherit styles without explicit props. The problem I found was that wrapping `<ShowMore>` in e.g. styled `<Text>` would also result in those styles being applied to all `children` components. 

Not the end of the world, but means anywhere this is used the children would likely have to specify their own styles to override the styles we only want applied to `ShowMore`, which is a headache.

This approach removes the styles from `ShowMore` itself while allowing parents to pass in styles without an explicit `clickableProps/textProps` interface. It's a little odd because any props shared by both sets of props get passed down to both `Text` and `Clickable` - if someone were to do something like add `mt` to the list of allowed `Text` props in `splitTextProps`, then we'd suddenly start doubling the top margin everywhere we pass one in.

Reply:
> if someone were to do something like add mt to the list of allowed Text props in splitTextProps, then we'd suddenly start doubling the top margin everywhere we pass one in.

That makes it a no go then — you could just split off `variant` without bothering with `splitProps` and everything else goes to the container?

I'm also willing to be pragmatic if this is holding up something — but I'd really recommend setting up a story that accomplishes your use-case with the comma insertion because looking at it I think it's probably going to be more cumbersome than just some one off component in Force.

Reply:
>That makes it a no go then — you could just split off variant without bothering with splitProps and everything else goes to the container?

Oh yeah, good call! Updated, that looks better and simplifies things a lot.

> I'd really recommend setting up a story that accomplishes your use-case with the comma insertion because looking at it I think it's probably going to be more cumbersome than just some one off component in Force.

Yep this PR includes it already! And yeah, this was more meant to be an improvement to the component than a time-saver. Just trying to do it the "right" way, but turns out I'm a bit out of touch with best patterns in Palette 😅 thanks for helping me navigate this.

![Screen Shot 2022-08-30 at 17 40 34](https://user-images.githubusercontent.com/5361806/187480879-200c31ec-e6de-4c23-8249-eec5b0430bf5.png)